server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -964,13 +964,7 @@ public PublicIp doInTransaction(TransactionStatus status) throws InsufficientAdd
                         VpcVO vpc = _vpcDao.findById(vpcId);
                         displayIp = vpc.isDisplay();
                     }
-                    PublicIp ip = fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, false, null, false, vpcId, displayIp, false);
-                    IPAddressVO publicIp = ip.ip();
-
-                    markPublicIpAsAllocated(publicIp);
-                    _ipAddressDao.update(publicIp.getId(), publicIp);
-
-                    return ip;
+                    return fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, true, null, false, vpcId, displayIp, false);
```

Comment:
this is a revert of the code in #2295. Is that functionality now broken?

Reply:
Yes, that PR incorrectly addressed merge conflict. One of the booleans changed from true to false, this fixes that.